contracts/core/Registry.sol

```diff
@@ -252,12 +251,18 @@ contract Registry is IRegistry, Native, AccessControl, Transfer {
     /// ======== Internal Functions ========
     /// ====================================
 
-    /// @notice Generates the anchor for the given identityId and name
+    /// @notice Generates and deploy the anchor for the given identityId and name
     /// @param _identityId Id of the identity
     /// @param _name The name of the identity
-    function _generateAnchor(bytes32 _identityId, string memory _name) internal pure returns (address) {
-        bytes32 attestationHash = keccak256(abi.encodePacked(_identityId, _name));
-        return address(uint160(uint256(attestationHash)));
+    function _generateAnchor(bytes32 _identityId, string memory _name) internal returns (address) {
+        bytes32 salt = keccak256(abi.encodePacked(_identityId, _name));
```

Comment:
This is awesome because it's replicable across chains. Same nonce and original msg.sender for creating the identity, same name, and they'll automatically get the same salt. We just need to make sure Registry.sol is at the same address on all chains (which shouldn't be a problem using CREATE2).